includes/class-migration.php

```diff
@@ -337,4 +341,31 @@ private static function update_options_key( $old, $new ) { // phpcs:ignore
 			array( '%s' )
 		);
 	}
+
+	/**
+	 * Migrate from 3.X
+	 *
+	 * * Get the ID of the latest blog post and save it to the options table
+	 *
+	 * @return void
+	 */
+	private static function migrate_from_3_x() {
+		$latest_post_id = 0;
+
+		// get the ID of the latest blog post and save it to the options table
+		$latest_post = get_posts(
+			array(
+				'numberposts' => 1,
+				'orderby'     => 'date',
+				'order'       => 'DESC',
+				'post_status' => 'publish',
+			)
+		);
+
+		if ( $latest_post ) {
+			$latest_post_id = $latest_post[0]->ID;
+		}
+
+		update_option( 'activitypub_last_legacy_post', $latest_post_id );
```

Comment:
Suggestion: maybe choose a very clear option name like: `activitypub_last_post_with_pretty_permalink_as_id`

Reply:
I thought that might be too long (49 chars) so I double-checked: `option_name` is `varchar(191)` and we're just fine :)